source/MenuPanel.cpp

```diff
@@ -94,6 +94,17 @@ MenuPanel::MenuPanel(PlayerInfo &player, UI &gamePanels)
 
 	if(!scrollSpeed)
 		scrollSpeed = 1;
+
+	// When the player is in the menu, pause the game sounds.
+	if(pauseAudio)
+		Audio::Pause();
+}
+
+
+
+MenuPanel::~MenuPanel()
+{
+	Audio::Resume();
```

Comment:
It seems strange to have `Audio::Pause()` controlled by a parameter in the constructor of the class, and to have `Audio::Resume()` just always executed. If pausing is optional, then I would expect resuming to also be optional (controlled through the same parameter), and if pausing just always need to happen, then don't make it a parameter, but just perform it unconditionally.

Reply:
Resuming is unconditional because it only affects paused streams - if we didn't pause anything, it is effectively a no-op.

I could make it conditional, but there is no real need to, at present. The only case where this would break something is if we had paused audio before creating this panel, which doesn't happen here.

With that said, it is probably more future-proof to have it be conditional, though.

Reply:
No need to make it optional, just always unpausing audio unconditionally seems fine (as long as we have only 1 MenuPanel, which is the case at the moment).